client/server/pages/index.js

```diff
@@ -476,35 +476,58 @@ function setUpLoggedInRoute( req, res, next ) {
 }
 
 /**
- * Sets up a Content Security Policy header
+ * Sets up a Content Security Policy header for all Calypso routes
+ *
+ * This CSP is currently in REPORT-ONLY mode, which means violations are logged but not blocked.
+ * This allows us to identify issues before enforcing the policy.
+ *
+ * Security approach:
+ * - Uses cryptographic nonces for inline scripts (generated per request)
+ * - Allows specific trusted third-party domains (payment providers, analytics, fonts)
+ * - Environment-aware: allows 'unsafe-eval' only in dev (for webpack eval source maps)
+ * - Reports violations to /cspreport endpoint for monitoring
+ *
+ * Required for compliance on pages handling credit card information.
  * @see https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP
  * @param {Object} req Express request object
  * @param {Object} res Express response object
  * @param {Function} next a callback to call when done
  */
 function setUpCSP( req, res, next ) {
-	const originalUrlPathname = req.originalUrl.split( '?' )[ 0 ];
-
-	// We only setup CSP for /log-in* for now
-	if ( ! /^\/log-in/.test( originalUrlPathname ) ) {
-		next();
-		return;
-	}
+	// CSP is now applied to all routes for security compliance (previously only /log-in*).
+	// This is necessary because Calypso is an SPA - the initial page load's CSP applies
+	// to the entire session, so we need CSP protection on all entry points, especially
+	// pages that handle credit card information (checkout, payment methods, billing).
 
 	// This is calculated by taking the contents of the script text from between the tags,
 	// and calculating SHA256 hash on it, encoded in base64, example:
 	// `sha256-${ base64( sha256( 'window.AppBoot();' ) ) }` === sha256-3yiQswl88knA3EhjrG5tj5gmV6EUdLYFvn2dygc0xUQ
 	// you can also just run it in Chrome, chrome will give you the hash of the violating scripts
-	const inlineScripts = [ 'sha256-ZKTuGaoyrLu2lwYpcyzib+xE4/2mCN8PKv31uXS3Eg4=' ];
+	const inlineScripts = [
+		'sha256-ZKTuGaoyrLu2lwYpcyzib+xE4/2mCN8PKv31uXS3Eg4=',
+		'sha256-Ab4XY87C+5uFAaNDtQ+4UTgs8t6MTnQmMtSsZCvmoiw=', // Additional inline script
+		'sha256-ehBD9wGNfnN0flaZIjbVClW1//FJFsATigcdVB4VdMQ=', // Additional inline script
+	];
 
 	req.context.inlineScriptNonce = crypto.randomBytes( 48 ).toString( 'hex' );
 
+	// Development environments that need webpack eval-based source maps
+	const isDevelopmentEnv = [
+		'development',
+		'wpcalypso',
```

Comment:
Is `wpcalypso` really a development environment? It uses the same production build as `production`, just with different feature flags. Or is this because of devdocs?